CHANGELOG.md

```diff
@@ -5,7 +5,8 @@
 ### Fixes
 
 - Relocate fee sanity check and make it relative to each transaction (rather than absolute) ([CO-446](https://iohk.myjetbrains.com/youtrack/issue/CO-446) [#3993](https://github.com/input-output-hk/cardano-sl/pull/3993)
-
+- Change how block retrieval worker's conversations are organised, so that the worker can recover from IOExceptions ([CDEC-658](https://iohk.myjetbrains.com/youtrack/issue/CDEC-658) [#3994](https://github.com/input-output-hk/cardano-sl/pull/3994))
+- Limit the rate of dns queries rate to avoid reaching the limit of open files, try resolve dns in ntp client every 30s rather than once ([CDEC-659](https://iohk.myjetbrains.com/youtrack/issue/CDEC-658) [#4008](https://github.com/input-output-hk/cardano-sl/pull/4008))
```

Comment:
Limit the rate of dns queries ~rate~ to avoid reaching the limit of open files; try to resolve dns in ntp client every 30s rather than *only* once.

Reply:
I updated the PR.